src/main/java/com/ericsson/gerrit/plugins/eiffel/handlers/StateHandler.java

```diff
@@ -0,0 +1,207 @@
+    }
+
+    /**
+     * This function returns the last set scm change eiffel event id stored based on
+     * the branch and specific project.
+     *
+     * @param project
+     * @param branch
+     * @return
+     */
+    public String getLastSourceChangeSubmittedEiffelEvent(String project, String branch) {
+        if (isDisabled()) {
+            return "";
+        }
+
+        File parentDir = new File(buildParentFilePath(project));
+        if(!(parentDir.exists())) {
+            return "";
+        }
+
+        try {
+            String fileName = String.format("%s.%s", project, FILE_ENDING);
+            DataBaseHandler dBHandler = new DataBaseHandler(pluginDir, fileName);
+            String eventId = dBHandler.getEventID(Table.SCS_TABLE, branch);
+            LOGGER.info("Fetched old event with id '{}', for project '{}', and branch '{}'", eventId, project, branch);
+            return eventId;
+        } catch (Exception e) {
+            LOGGER.error("Error while trying to get eiffel event id from database: {}\n{}", e.getMessage(), e);
+            return "";
+        }
+
+    }
+
+    /**
+     * This function sets an scm change eiffel event id for a specific project and
+     * branch.
+     *
+     * @param project
+     * @param branch
+     * @param eiffelEvent
+     */
+    public void setLastSourceChangeSubmittedEiffelEvent(String project, String branch, String eiffelEvent) {
+        if (isDisabled()) {
+            return;
+        }
+
+        DataBaseHandler dBHandler;
+        try {
+            String parentPath = buildParentFilePath(project);
+            createParentDirsIfNecessary(parentPath);
+
+            String fileName = String.format("%s.%s", project, FILE_ENDING);
+            dBHandler = new DataBaseHandler(pluginDir, fileName);
+            String oldEvent = dBHandler.getEventID(Table.SCS_TABLE, branch);
+            if (!oldEvent.isEmpty()) {
+                dBHandler.updateInto(Table.SCS_TABLE, branch, eiffelEvent);
+                LOGGER.info("Replaced old event id '{}' with new event if '{}', for project '{}', and branch '{}'.",
+                        oldEvent, eiffelEvent, project, branch);
+            } else {
+                dBHandler.insertInto(Table.SCS_TABLE, branch, eiffelEvent);
+                LOGGER.info("Saved eiffel event with id '{}', for project '{}', and branch '{}'.", eiffelEvent, project,
+                        branch);
+            }
+        } catch (Exception e) {
+            LOGGER.error("Error while trying to insert eiffel event id into database: {}\n{}", e.getMessage(), e);
+        }
+
+    }
+
+    /**
+     * Builds the absolute file path to the parent of a project
+     *
+     * @param project
+     * @return
+     */
+    private String buildParentFilePath(String project) {
+        int lastIndexOfSlash = project.lastIndexOf("/");
+
+        String relativeParentPath = "";
+        boolean projectContainsParent = lastIndexOfSlash != -1;
```

Comment:
Instead of breaking out to a variable, break out to a method. In that way, you don't need to see the logic in this code. Only if the reader wants to.

Reply:
done